### PR TITLE
Fixed zoom for map

### PR DESCRIPTION
**Pull request by smialko**
**Task:** https://trello.com/c/jv3Hk6EG/227-mobile-map-improve-mapping-visualization

Added autozoom based on what Android has, it searches for the most extreme long/lat coordinates, adds some padding, and displays the map accordingly. 
Although it's not entirely fixed, I'd like to send a build for Michael to test.

### CODE REVIEW

File: AirCasting/Map/GoogleMapView.swift [hunk 1]
```diff
@@ -60,43 +60,69 @@ struct GoogleMapView: UIViewRepresentable {
...
+        let polyline = context.coordinator.polyline
         let spans = pathPoints.map { point -> GMSStyleSpan in
             let color = colorPolyline(point: point)
             return GMSStyleSpan(style: GMSStrokeStyle.solidColor(color),
                                 segments: 1)
         }
+
+        polyline.path = path
+        polyline.spans = spans
+        polyline.strokeWidth = 10
```
Comment (jankrzempek) [inline AirCasting/Map/GoogleMapView.swift:72]: Do we want to add this to `Constants.swift` or something or no 😄?

File: AirCasting/Map/GoogleMapView.swift [hunk 2]
```diff
@@ -60,43 +60,69 @@ struct GoogleMapView: UIViewRepresentable {
...
-            context.coordinator.didSetInitLocation = true
         }
-        polyline.spans = spans
-        polyline.strokeWidth = 6
-        polyline.map = uiView
+    }
+    
+    var cameraUpdate: GMSCameraUpdate {
+        guard !pathPoints.isEmpty else {
+            let location = tracker.googleLocation.last?.location ?? CLLocationCoordinate2D(latitude: 0, longitude: 0)
```
Comment (jankrzempek) [inline AirCasting/Map/GoogleMapView.swift:86]: why CLLocationCoordinate2D(latitude: 0, longitude: 0)? I think we have FakeLocation somewhere

File: AirCasting/Map/GoogleMapView.swift [hunk 3]
```diff
@@ -60,43 +60,69 @@ struct GoogleMapView: UIViewRepresentable {
...
+            return GMSCameraUpdate.setTarget(location)
+        }
+        
+        var north = -90.0
+        var south = 90.0
+        var east = -180.0
+        var west = 180.0
```
Comment (jankrzempek) [inline AirCasting/Map/GoogleMapView.swift:93]: Constants or no?

**Review by jankrzempek — APPROVED** [review]
Cool 🥳

File: AirCasting/Map/GoogleMapView.swift (hunk 1 above)
Comment (smialko) [reply AirCasting/Map/GoogleMapView.swift:72]: Yep, good idea

File: AirCasting/Map/GoogleMapView.swift (hunk 3 above)
Comment (smialko) [reply AirCasting/Map/GoogleMapView.swift:93]: I'm not sure about that, these are just initial values and we change them right after that, I don't think they fall into the "Constats" category.

File: AirCasting/Map/GoogleMapView.swift (hunk 2 above)
Comment (smialko) [reply AirCasting/Map/GoogleMapView.swift:86]: Yes, we do in CreateSessionDetailsView it's not global though.

File: AirCasting/Map/GoogleMapView.swift [hunk 4]
```diff
@@ -60,43 +60,69 @@ struct GoogleMapView: UIViewRepresentable {
...
+                let longitude = point.location.longitude
+                
+                north = max(north, latitude)
+                south = min(south, latitude)
+                east = max(east, longitude)
+                west = min(west, longitude)
+            }
+        
+        let southWest = CLLocationCoordinate2D(latitude: south, longitude: west)
+        let northEast = CLLocationCoordinate2D(latitude: north, longitude: east)
```
Comment (pawelgil) [inline AirCasting/Map/GoogleMapView.swift:107]: as a challenge you can try to do it all using `pathPoints.reduce`

**Review by pawelgil — APPROVED** [review]

**Review by jankrzempek — APPROVED** [review]
🥳🥳

File: AirCasting/Map/GoogleMapView.swift (hunk 4 above)
Comment (smialko) [reply AirCasting/Map/GoogleMapView.swift]: Challenge accepted 😎

I know you already approved but could you please take a look if I did it right?

**Comment by smialko** [comment]
I'm merging it bc we want to send a build but it would be awesome if @pawelgil would check if I refactored it right :D

**Pull request merged**